crates/chain/src/local_chain.rs

```diff
@@ -148,6 +149,34 @@ impl CheckPoint {
     pub fn iter(&self) -> CheckPointIter {
         self.clone().into_iter()
     }
+
+    /// Find checkpoint at `height`.
+    ///
+    /// Returns `None` if checkpoint at `height` does not exist`. If you are querying for multiple
+    /// heights, it will be more efficient to use [`query_multi`].
+    ///
+    /// [`query_multi`]: CheckPoint::query_multi
+    pub fn query(&self, height: u32) -> Option<Self> {
+        self.iter()
+            .take_while(|cp| cp.height() >= height)
+            .find(|cp| cp.height() == height)
+    }
+
+    /// Find checkpoints of the given `heights`.
+    ///
+    /// This returns an iterator which iterates over the requested heights and whether a
+    /// corresponding checkpoint is found.
+    pub fn query_multi(&self, heights: BTreeSet<u32>) -> impl Iterator<Item = (u32, Option<Self>)> {
```

Comment:
Seems like a premature optimization. I understand that this is faster than calling query several times but the way to speed `query` up is with a skip list.

Reply:
Even with a skip list, this will be faster? This is also a convenience method, like super convenient. I honestly think this is worth it.

Reply:
> Even with a skip list, this will be faster? This is also a convenience method, like super convenient. I honestly think this is worth it.

Things that are faster are not necessarily better. Can we add the convenience method at the point that we need the convenience?

I think in general this PR doesn't really belong at this point because we don't get to see how we actually use the API. It seems like a simple thing to do as part of a larger PR that deletes the BTreeMap from LocalChain and uses checkpoint queries instead. Can we just make that this PR?

Reply:
I realized that you are right. I think it is a great policy to only introduce convenience methods if they are used. `CheckPoint::query_multi` turns out to be not very useful. However, I introduced `CheckPoint::query_from` which will return `Some` if there is a checkpoint at `height` or greater. This is a sufficient replacement for `BTreeMap::<u32, BlockHash>::range` which is used in `bdk::Wallet::insert_tx`.